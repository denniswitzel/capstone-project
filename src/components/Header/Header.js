import React from 'react'
import styled from 'styled-components'
import { ReactComponent as Arrow } from '../../images/icons/back-arrow.svg'
import { ReactComponent as Filter } from '../../images/icons/filter.svg'
import { useHistory } from 'react-router-dom'
import PropTypes from 'prop-types'

Headline.propTypes = {
  headline: PropTypes.string,
  onToggle: PropTypes.func,
  active: PropTypes.string
}

export default function Headline({ headline, onToggle, active }) {
  const history = useHistory()
  function handleClick() {
    history.goBack()
  }

  return (
    <Header>
      <ArrowStyled onClick={handleClick} />
      <HeadlineStyled>{headline}</HeadlineStyled>
      <FilterStyled active={active} onClick={onToggle}/>
    </Header>
  )
}

const Header = styled.section`
  display: flex;
  grid-column: 1/3;
  margin: 20px 0;
  align-items: center;
  max-width: 400px;
`

const HeadlineStyled = styled.h2`
  display: inline;
  font-weight: 300;
  color: var(--grey);
  border-bottom: 1px solid var(--green);
  padding-bottom: 5px;
  margin: 0 auto;
  max-width: 300px;
`
const ArrowStyled = styled(Arrow)`
  fill: var(--grey);
  position: absolute;
  left: 8%;
  cursor: pointer;
`
const FilterStyled = styled(Filter)`
  position: absolute;
  right: 8%;
  cursor: pointer;
  stroke: ${({active}) => active === 'true' ? 'var(--green)' : 'var(--grey)'};
  fill: transparent;
`
