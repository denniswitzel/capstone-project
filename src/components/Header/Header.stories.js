import React from 'react';
import Header from './Header'

export default {
  title: 'Components/Header',
  component: Header,
  argTypes: {
    headline: { control: {
      type: 'select',
      options: ['Burger', 'Cold Cuts', 'Kebab', 'Minced', 'Nuggets', 'Sausages']
    }}
  },
};

const Template = (args) => <Header {...args} />;

export const Headline = Template.bind({});
Headline.args = {
  headline: 'Burger'
}