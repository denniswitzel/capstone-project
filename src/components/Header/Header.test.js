import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Header', () => {
  const headline = 'BURGER'

  it('display the headline and an example event', () => {
    const { getByText } = render(<Header headline={headline} />)
    expect(getByText(headline)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(<Header />)
    expect(tree).toMatchSnapshot()
  })
})
