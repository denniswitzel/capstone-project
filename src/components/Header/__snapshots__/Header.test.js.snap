// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<section
  className="sc-AxjAm jRGLJZ"
>
  <svg
    className="sc-AxiKw jcUDjX"
    onClick={[Function]}
  >
    back-arrow.svg
  </svg>
  <h2
    className="sc-AxirZ jgYDSb"
  />
  <svg
    className="sc-AxhCb FeAGO"
  >
    filter.svg
  </svg>
</section>
`;
